lib/private/Share20/Manager.php

```diff
@@ -379,28 +307,53 @@ protected function validateExpirationDateInternal(IShare $share) {
 			$configProp = 'internal_defaultExpDays';
 			$isEnforced = $this->shareApiInternalDefaultExpireDateEnforced();
 		}
-		if ($fullId === null && $expirationDate === null && $defaultExpireDate) {
-			$expirationDate = new \DateTime('now', $this->dateTimeZone->getTimeZone());
-			$expirationDate->setTime(0, 0, 0);
-			$days = (int)$this->config->getAppValue('core', $configProp, (string)$defaultExpireDays);
-			if ($days > $defaultExpireDays) {
-				$days = $defaultExpireDays;
+
+		// If $expirationDate is falsy, noExpirationDate is true and expiration not enforced
+		// Then skip expiration date validation as null is accepted
+		if(!(empty($expirationDate) && $share->getNoExpirationDate() && !$isEnforced)) {
```

Comment:
Does not make a difference, would update.

```suggestion
		if(!($share->getNoExpirationDate() && !$isEnforced)) {
```